### PR TITLE
Standalone OSX Build Fixes

### PR DESCRIPTION
**Pull request by dylwhich**
### Description

Contains standalone fixes for OSX from the midi-synth-platform-midi branch.

### Test Instructions

Tested on the midi-synth-platform-midi branch and with an updated workflow that builds for OSX also.

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: .github/workflows/build-firmware-and-emulator.yml [hunk 1]
```diff
@@ -14,58 +14,125 @@ env:
 
 jobs:
   Build-Firmware-And-Emulator:
-    runs-on: windows-latest
+    strategy:
+      matrix:
+        os: [ windows, osx-arm, osx-intel ]
```
Comment (AEFeinstein) [inline .github/workflows/build-firmware-and-emulator.yml:19]: Wanna throw a Linux build in here for kicks?

Comment (AEFeinstein) [reply .github/workflows/build-firmware-and-emulator.yml:19]: Sidebar, Mac builds are reasonably speedy. Nice. msys2 and the IDF are killers. We should probably be [doing some caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows).

File: makefile [hunk 2]
```diff
@@ -96,8 +95,10 @@ ifneq ($(HOST_OS),Darwin)
 # Incompatible flags for clang on MacOS
 CFLAGS += \
 	-static-libgcc \
+	-static-libstdc++ \
 	-fstrict-volatile-bitfields \
-	-fno-tree-switch-conversion
+	-fno-tree-switch-conversion \
+	-fno-omit-frame-pointer
```
Comment (AEFeinstein) [inline makefile:101]: Linux gets this twice? No harm, no foul.

**Review by AEFeinstein — APPROVED** [review]
LGTM. I may be getting greedy with Github Actions improvements, but now that you've touched the YAML, I can pester you about it 😄

File: .github/workflows/build-firmware-and-emulator.yml (hunk 1 above)
Comment (dylwhich) [reply .github/workflows/build-firmware-and-emulator.yml:19]: > Wanna throw a Linux build in here for kicks?

Ah yeah, might as well!

Comment (dylwhich) [reply .github/workflows/build-firmware-and-emulator.yml:19]: > Sidebar, Mac builds are reasonably speedy. Nice. msys2 and the IDF are killers. We should probably be [doing some caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows).

Hmm, I took a look but it looks like all those are just for language-level package managers like node and stuff. But! I did find this action which claims to support caching, I'll give it a try: https://github.com/msys2/setup-msys2

Comment (dylwhich) [reply .github/workflows/build-firmware-and-emulator.yml]: Nice, Linux builds in under 30 seconds! :sunglasses:

I tried the msys2 setup stuff, but it does a bunch of extra stuff I don't really understand and fails. I'll try changing the `MSYSTEM` config and see if that helps but if it fails again I'll just leave it for later.

Comment (dylwhich) [reply .github/workflows/build-firmware-and-emulator.yml]: Okay, just caching the IDF knocks about 5 minutes off the Windows build so now it's done in a _blazing_ ten minutes!

File: makefile (hunk 2 above)
Comment (dylwhich) [reply makefile:101]: Ah, I took it out of the other spot anyway.

**Pull request merged**